File: backend/alpha_dklg/src/main/java/com/portfolio/alpha_dklg/service/SkillService.java
package com.portfolio.alpha_dklg.service;

import com.portfolio.alpha_dklg.model.Skill;
import java.util.List;

public interface SkillService {
    Skill saveSkill(Skill skill);
    Skill getSkillById(Long id);
    List<Skill> getAllSkills();
    Skill updateSkill(Long id, Skill skill);
    void deleteSkill(Long id);
    
    // Filter by category
    List<Skill> getSkillsByCategory(String category);
} 

File: frontend/alpha_dklg/src/app/services/skill.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Skill } from '../models/skill.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class SkillService {
  private apiUrl = `${environment.apiUrl}/skills`;

  constructor(private http: HttpClient) {}

  getAllSkills(): Observable<Skill[]> {
    return this.http.get<Skill[]>(this.apiUrl);
  }

  getSkillById(id: number): Observable<Skill> {
    return this.http.get<Skill>(`${this.apiUrl}/${id}`);
  }

  getSkillsByCategory(category: string): Observable<Skill[]> {
    return this.http.get<Skill[]>(`${this.apiUrl}/category/${category}`);
  }

  createSkill(skill: Omit<Skill, 'id'>): Observable<Skill> {
    return this.http.post<Skill>(this.apiUrl, skill);
  }

  updateSkill(id: number, skill: Omit<Skill, 'id'>): Observable<Skill> {
    return this.http.put<Skill>(`${this.apiUrl}/${id}`, skill);
  }

  deleteSkill(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
} 